cover_agent/CoverageProcessor.py

```diff
@@ -116,3 +122,63 @@ def parse_coverage_report_cobertura(self) -> Tuple[list, list, float]:
         )
 
         return lines_covered, lines_missed, coverage_percentage
+
+    def parse_coverage_report_jacoco(self) -> Tuple[list, list, float]:
+        """
+        Parses a JaCoCo XML code coverage report to extract covered and missed line numbers for a specific file,
+        and calculates the coverage percentage.
+
+        Returns: Tuple[list, list, float]: A tuple containing empty lists of covered and missed line numbers,
+        and the coverage percentage. The reason being the format of the report for jacoco gives the totals we do not
+        sum them up. to stick with the current contract of the code and to do little change returning empty arrays.
+        I expect this should bring up a discussion on introduce a factory for different CoverageProcessors. Where the
+        total coverage percentage is returned to be evaluated only.
+        """
+        lines_covered, lines_missed = [], []
+
+        package_name, class_name = self.extract_package_and_class()
+        missed, covered = self.parse_missed_covered_lines(package_name, class_name)
+
+        total_lines = missed + covered
+        coverage_percentage = (covered / total_lines) if total_lines > 0 else 0
+
+        return lines_covered, lines_missed, coverage_percentage
+
+    def parse_missed_covered_lines(self, package_name: str, class_name: str) -> tuple[int, int]:
```

Comment:
Is this `jacoco` specific? If so, it may be prudent to append `jacoco` to the end of the function name. 

If we start to support more coverage types we may want to move coverage processors to a dedicated folder and have coverage types as their own class that inherit and override functions from a base class.